tools/libgcopdd.py

```diff
@@ -0,0 +1,104 @@
+import sys
+import os
+from os import path
+import gzip
+
+
+class Event:
+	def __init__(self, event_id=None, **loaded_entries):
+		self.event_id = event_id
+		self.loaded_entries = loaded_entries
+	
+	def get(self, entry_type, load):
+		result = self.loaded_entries.get(entry_type)
+		if result is None:
+			self.loaded_entries[entry_type] = result = load(self.event_id)
+		return result
+
+
+class UserError(Exception):
+	def __init__(self, msg):
+		self.msg = msg
+
+
+def args_to_events(args):
+	def inner():
+		for arg in args:
+			if path.isdir(arg):
+				# this is a dump
+				for basename in os.listdir(arg):
+					event_id, entry_type, *_ = basename.split('.')
+					if entry_type == 'request':
+						yield arg + '/' + event_id
+			elif path.exists(arg):
+				# this is an event list
+				with open(arg) as file:
+					for line in file:
+						line = line.split('#')[0].strip()
+						if line != '':
+							yield line
+			else:
+				# this is probably a single event
+				yield arg
+	return list(map(Event, inner()))
+
+
+def n_args_to_n_events(n, args, msg):
+	if len(args) != n:
```

Comment:
if there is such condition, why is the n required as an argument. Can'ŧ we just use len(args)?